src/main/java/com/unstoppabledomains/resolution/contracts/uns/Registry.java

```diff
@@ -0,0 +1,39 @@
+package com.unstoppabledomains.resolution.contracts.uns;
+
+import com.esaulpaugh.headlong.abi.Tuple;
+import com.unstoppabledomains.resolution.contracts.BaseContract;
+import com.unstoppabledomains.resolution.contracts.interfaces.IProvider;
+import com.unstoppabledomains.util.Utilities;
+
+import java.math.BigInteger;
+import java.util.List;
+
+public class Registry extends BaseContract {
+
+    private static final String ABI_FILE = "cns/registry_abi.json";
+    private static final String namingServiceName = "CNS";
+
+    public Registry(String url, String address, IProvider provider) {
+        super(namingServiceName, url, address, provider);
+    }
+
+    public String getDomainName(BigInteger tokenID) {
+        try {
+            String[] args = { Utilities.tokenIDToNamehash(tokenID) };
+            List<Tuple> logs = fetchLogs("earliest", "NewURI", args);
```

Comment:
@DancingAxolotl Good idea. If Infura allow to use `earliest` as a block number we should use it in other libraries and avoid using deploymentBlock from config.

Reply:
It works, but I'm not sure if it is using the deployment block or just the earliest block in the network.